source/cell/cell.cpp

```diff
@@ -802,8 +828,11 @@ void cell::value(const std::string &value_string, bool infer_type)
 
 void cell::clear_format()
 {
-    format().d_->references -= format().d_->references > 0 ? 1 : 0;
-    d_->format_.clear();
+    if (d_->format_.is_set())
+    {
+        format().d_->references -= format().d_->references > 0 ? 1 : 0;
```

Comment:
A recommendation: don't use unsigned numbers if you don't have to. While it may seem sensible to use them to represent "only positive values", they have too many edge cases that are easy to not consider.

Unsigned integers are better reserved for bitwise operations (where signed values exhibit undefined behaviour), and using signed values for counting/maths